balancer-js/src/modules/data/pool/subgraph/poolDataService.ts

```diff
@@ -36,7 +36,9 @@ export class SubgraphPoolDataService implements PoolDataService {
   }
 
   public async getPools(): Promise<SubgraphPoolBase[]> {
-    const pools = await this.subgraphHelper.allPools();
+    const pools = await this.subgraphHelper.allPools({
+      poolsToIgnore: this.network.poolsToIgnore,
```

Comment:
can we set it on the default query instead as something like?

```js
where: {
  address: {
    not_in: this.network.poolsToIgnore
  }
}
```

Reply:
See the [PR/Asana](https://github.com/balancer-labs/balancer-sdk/pull/267) for reasons to avoid query.

Reply:
let's bench it, it might be that `id_not_in` would be much faster

Reply:
All queries with `id_not_in` or `address_not_id` or without any filter have the same response times of ~600ms on avg.

Reply:
why it's so hard to make subgraph perfect.. IIRC Mike noticed slow queries for swapEnabled and minLiquidity. asking Markus if they are indexed, so it doesn't hurt other queries like that. if that turns out absolutely hopeless then lets patch in JS.

Reply:
at the end whatever works